test/resilience/tests/envoyproxy.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+//go:build resilience
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"github.com/envoyproxy/gateway/test/resilience/suite"
+	"github.com/go-logr/zapr"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"testing"
+	"time"
+)
+
+func init() {
+	ResilienceTests = append(ResilienceTests, EPResilience)
+	zapLog, err := zap.NewDevelopment() // Use zap.NewProduction() for production
+	if err != nil {
+		panic(fmt.Sprintf("Failed to create logger: %v", err))
+	}
+	log.SetLogger(zapr.NewLogger(zapLog))
```

Comment:
why we need this here?

Reply:
its Integrating the Kubernetes client (client-go) with the zap logging library enhances the visibility of internal operations and API interactions, facilitating more effective debugging and monitoring. so incase things fail, its helps to show some of internal client errors. however we can skip that.

Reply:
it should be in main or somewhere, better not to be in `init` like https://github.com/envoyproxy/gateway/blob/0f68219d191b2b1d5953d59945c5e174db229c1c/test/e2e/e2e_test.go#L30

Reply:
indeed moved that.